google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbstractResultSet.java

```diff
@@ -679,6 +680,60 @@ protected Date getDateInternal(int columnIndex) {
       return (Date) rowData.get(columnIndex);
     }
 
+    @Override
+    protected Value getValueInternal(int columnIndex) {
+      final List<Type.StructField> structFields = getType().getStructFields();
+      final StructField structField = structFields.get(columnIndex);
+      final Type columnType = structField.getType();
+      switch (columnType.getCode()) {
+        case BOOL:
+          return Value.bool(getBooleanInternal(columnIndex));
+        case INT64:
+          return Value.int64(getLongInternal(columnIndex));
+        case NUMERIC:
+          return Value.numeric(getBigDecimalInternal(columnIndex));
+        case FLOAT64:
+          return Value.float64(getDoubleInternal(columnIndex));
+        case STRING:
+          return Value.string(getStringInternal(columnIndex));
+        case BYTES:
+          return Value.bytes(getBytesInternal(columnIndex));
+        case TIMESTAMP:
+          return Value.timestamp(getTimestampInternal(columnIndex));
+        case DATE:
+          return Value.date(getDateInternal(columnIndex));
+        case STRUCT:
+          return Value.struct(getStructInternal(columnIndex));
+        case ARRAY:
+          switch (columnType.getArrayElementType().getCode()) {
+            case BOOL:
+              return Value.boolArray(getBooleanArrayInternal(columnIndex));
+            case INT64:
+              return Value.int64Array(getLongArrayInternal(columnIndex));
+            case NUMERIC:
+              return Value.numericArray(getBigDecimalListInternal(columnIndex));
+            case FLOAT64:
+              return Value.float64Array(getDoubleArrayInternal(columnIndex));
+            case STRING:
+              return Value.stringArray(getStringListInternal(columnIndex));
+            case BYTES:
+              return Value.bytesArray(getBytesListInternal(columnIndex));
+            case TIMESTAMP:
+              return Value.timestampArray(getTimestampListInternal(columnIndex));
+            case DATE:
+              return Value.dateArray(getDateListInternal(columnIndex));
+            case STRUCT:
+              return Value.structArray(
+                  this.type.getArrayElementType(), getStructListInternal(columnIndex));
+            default:
```

Comment:
I assume you can't have arrays of arrays here.

Reply:
Correct, [arrays-of-arrays are not supported](https://cloud.google.com/spanner/docs/data-definition-language#arrays). Also, `ARRAY<STRUCT>` is not supported as the type of a table column, but can be [created in SQL statements](https://cloud.google.com/spanner/docs/arrays#querying_nested_arrays).

Reply:
Added tests for all possible literals I could think of (including `ARRAY<STRUCT>`)